doc/source/data/shuffling-data.rst

```diff
@@ -0,0 +1,225 @@
+~~~~~~~~~~~~~~~~
+
+To randomly shuffle all rows globally, call :meth:`~ray.data.Dataset.random_shuffle`.
+
+.. testcode::
+
+    import ray
+
+    ds = (
+        ray.data.read_images("s3://anonymous@ray-example-data/image-datasets/simple")
+        .random_shuffle()
+    )
+
+.. tip::
+
+    This is the slowest option for shuffle, and requires transferring data across
+    network between workers. This option achieves the best randomness among all options.
+
+.. _optimizing_shuffles:
+
+Advanced: Optimizing shuffles
+-----------------------------
+
+Shuffle operations are *all-to-all* operations where the entire Dataset must be materialized in memory before execution can proceed.
+Currently, these are:
+
+* :meth:`Dataset.groupby <ray.data.Dataset.groupby>`
+* :meth:`Dataset.random_shuffle <ray.data.Dataset.random_shuffle>`
+* :meth:`Dataset.sort <ray.data.Dataset.sort>`
+
+.. note:: This is an active area of development. If your Dataset uses a shuffle operation and you are having trouble configuring shuffle, 
+    `file a Ray Data issue on GitHub <https://github.com/ray-project/ray/issues/new?assignees=&labels=bug%2Ctriage%2Cdata&projects=&template=bug-report.yml&title=[data]+>`_. 
+
+When should you use global per-epoch shuffling?
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Use global per-epoch shuffling only if your model is sensitive to the
+randomness of the training data. Based on a
+`theoretical foundation <https://arxiv.org/abs/1709.10432>`__ all
+gradient-descent-based model trainers benefit from improved (global) shuffle quality.
+In practice, the benefit is particularly pronounced for tabular data/models.
+However, the more global the shuffle is, the more expensive the shuffling operation.
+The increase compounds with distributed data-parallel training on a multi-node cluster due
+to data transfer costs. This cost can be prohibitive when using very large datasets.
+
+The best route for determining the best tradeoff between preprocessing time and cost and
+per-epoch shuffle quality is to measure the precision gain per training step for your
+particular model under different shuffling policies:
+
+* no shuffling,
+* local (per-shard) limited-memory shuffle buffer,
+* local (per-shard) shuffling,
+* windowed (pseudo-global) shuffling, and
+* fully global shuffling.
+
+As long as your data loading and shuffling throughput is higher than your training throughput, your GPU should
+be saturated. If you have shuffle-sensitive models, push the
+shuffle quality higher until this threshold is hit.
+
+.. _shuffle_performance_tips:
+
+Enabling push-based shuffle
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Some Dataset operations require a *shuffle* operation, meaning that data is shuffled from all of the input partitions to all of the output partitions.
+These operations include :meth:`Dataset.random_shuffle <ray.data.Dataset.random_shuffle>`,
+:meth:`Dataset.sort <ray.data.Dataset.sort>` and :meth:`Dataset.groupby <ray.data.Dataset.groupby>`.
```

Comment:
It's not super intuitive why these operations require a shuffle, if possible maybe add a quick sentence explaining?